spec/services/twilio_service_spec.rb

```diff
@@ -112,51 +104,26 @@
     end
 
     it 'uses a different Twilio account for different instances', sms: true do
-      expect(Rails.application.secrets).to receive(:twilio_accounts).
+      allow(Figaro.env).to receive(:twilio_accounts).
         and_return(
-          [{
-            'sid' => 'sid1',
-            'auth_token' => 'token1',
-            'number' => '1111111111'
-          }],
-          [{
-            'sid' => 'sid2',
-            'auth_token' => 'token2',
-            'number' => '2222222222'
-          }])
+          '[
+             {"sid":"sid1", "auth_token":"token1", "number":"1111111111"},
+             {"sid":"sid2", "auth_token":"token2", "number":"2222222222"}
+           ]'
+        )
 
       expect(Twilio::REST::Client).to receive(:new).with('sid1', 'token1').and_call_original
+      expect(Twilio::REST::Client).to receive(:new).with('sid2', 'token2').and_call_original
 
       TwilioService.new.send_sms(
         to: '5555555555',
         body: '!!CODE1!!'
       )
 
-      expect(messages.size).to eq(1)
-      msg1 = messages.first
-      expect(msg1.from).to eq('+11111111111')
-      clear_messages
-
-      expect(Twilio::REST::Client).to receive(:new).with('sid2', 'token2').and_call_original
-
       TwilioService.new.send_sms(
         to: '6666666666',
         body: '!!CODE2!!'
       )
-
-      expect(messages.size).to eq(1)
-      msg2 = messages.first
-      expect(msg2.from).to eq('+12222222222')
-    end
-  end
-
-  describe '.random_account' do
-    context 'twilio_accounts has multiple entries' do
-      it 'randomly samples one of the accounts' do
-        expect(Rails.application.secrets.twilio_accounts).to receive(:sample)
-
-        TwilioService.random_account
-      end
```

Comment:
It was redundant. We already have a test above that checks that different instances use different accounts when multiple accounts are present. `.random_account` is not meant to be a public method, and private methods should not be tested directly.